File: packages/sdk/sdk-ruby/tests/models_test.rb
require "minitest/autorun"
require_relative "../lib/index"

class ModelsTest < Minitest::Test
  def test_list_models_preserves_availability_reason
    client = PhaseoSdk::Phaseo.new(
      api_key: "test",
      enable_deprecation_warnings: false
    )

    calls = []
    client.raw_client.define_singleton_method(:request) do |method:, path:, query: nil, headers: nil, body: nil|
      calls << [method, path, query, headers, body]
      {
        "ok" => true,
        "availability_mode" => "all",
        "models" => [
          {
            "id" => "openai/gpt-5-mini",
            "providers" => [
              {
                "api_provider_id" => "openai",
                "availability_status" => "coming_soon",
                "availability_reason" => "scheduled"
              }
            ]
          }
        ]
      }
    end

    response = client.list_models("availability" => "all")

    assert_equal "all", response["availability_mode"]
    assert_equal "coming_soon", response["models"][0]["providers"][0]["availability_status"]
    assert_equal "scheduled", response["models"][0]["providers"][0]["availability_reason"]
    assert_equal [["GET", "/models", {"availability" => "all"}, nil, nil]], calls
  end

  def test_parameter_support_highlights_invalid_value
    report = PhaseoSdk::Phaseo.check_parameter_support(
      {
        "id" => "openai/example",
        "endpoints" => [{
          "id" => "openai:responses", "endpoint" => "responses", "routable" => true, "status" => "active",
          "provider" => { "id" => "openai" },
          "capabilities" => { "parameters" => ["temperature"], "parameter_details" => { "temperature" => { "supported" => true, "minimum" => 0, "maximum" => 1 } } }
        }]
      },
      { "temperature" => 1.5 }
    )

    refute report[:ok]
    assert_equal "supported", report[:parameters][0][:status]
    assert_empty report[:parameters][0][:accepted_by]
  end
end

File: packages/sdk/sdk-php/tests/models_test.php
<?php
declare(strict_types=1);

require_once __DIR__ . "/../src/index.php";

use Phaseo\Sdk\Phaseo;

function assert_true(bool $condition, string $message): void
{
    if (!$condition) {
        throw new RuntimeException($message);
    }
}

final class FakeModelsClient extends \Phaseo\Gen\Client
{
    public array $calls = [];

    public function __construct()
    {
        parent::__construct("https://example.test");
    }

    public function request(string $method, string $path, ?array $query = null, ?array $headers = null, $body = null)
    {
        $this->calls[] = [$method, $path, $query, $headers, $body];
        return [
            "ok" => true,
            "availability_mode" => "all",
            "models" => [
                [
                    "id" => "openai/gpt-5-mini",
                    "providers" => [
                        [
                            "api_provider_id" => "openai",
                            "availability_status" => "coming_soon",
                            "availability_reason" => "scheduled",
                        ],
                    ],
                ],
            ],
        ];
    }
}

$client = new Phaseo(
    apiKey: "test",
    basePath: "https://api.phaseo.app/v1",
    enableDeprecationWarnings: false
);

$fake = new FakeModelsClient();
$ref = new ReflectionClass($client);
$prop = $ref->getProperty("client");
$prop->setAccessible(true);
$prop->setValue($client, $fake);

$response = $client->listModels(["availability" => "all"]);
assert_true(($response["availability_mode"] ?? null) === "all", "expected availability_mode");
assert_true(($response["models"][0]["providers"][0]["availability_status"] ?? null) === "coming_soon", "expected availability_status");
assert_true(($response["models"][0]["providers"][0]["availability_reason"] ?? null) === "scheduled", "expected availability_reason");
assert_true($fake->calls === [
    ["GET", "/models", ["availability" => "all"], null, null],
], "expected request call for models discovery");

echo "php models tests ok" . PHP_EOL;
